edb/edgeql/compiler/inference/cardinality.py

```diff
@@ -775,6 +776,19 @@ def __infer_func_call(
                 span=ir.span,
             )
 
+    if ir.volatility == MODIFYING:
+        if any(card.is_multi() for card in cards):
+            raise errors.QueryError(
+                'possibly more than one element passed into modifying function',
+                span=ir.span,
+            )
+
+        if any(card.can_be_zero() for card in cards):
+            raise errors.QueryError(
+                'possibly an empty set passed into modifying function',
+                span=ir.span,
+            )
```

Comment:
Should work if the argument is OPTIONAL

Reply:
@msullivan Ah sorry, forgot to push the change.